src/librustc_metadata/locator.rs

```diff
@@ -53,6 +53,13 @@
 //! is a platform-defined dynamic library. Each library has a metadata somewhere
 //! inside of it.
 //!
+//! A third kind of dependency is an rmeta file. These are rlibs, which contain
```

Comment:
Having read through this file, everything looks good. I might expect though that `libfoo.rmeta` is not an archive but rather just the literal metadata file in the rlib normally itself. This'd avoid going through LLVM's archive support which may speed things up a bit, but probably won't have that much effect on perf.

I'm not sure if overall it'd be less code as well, just a thought!